samples/go/csv/csv-import/importer.go

```diff
@@ -146,15 +151,19 @@ func main() {
 		headers = strings.Split(*header, string(comma))
 	}
 
-	ds, err := spec.GetDataset(flag.Arg(dataSetArgN))
-	d.CheckError(err)
-	defer ds.Database().Close()
-
 	kinds := []types.NomsKind{}
 	if *columnTypes != "" {
 		kinds = csv.StringsToKinds(strings.Split(*columnTypes, ","))
+	} else if *detectColumnTypes {
+		kinds = csv.GetSchema(cr, headers)
```

Comment:
If I move it up to line 117 then I also need to move up (copy) or reimplement the creation of the actual csv.NewCSVReader, handling the skip records parameter, and the reading of the header line(s) depending on the params.  I do feel like this file is getting bigger and more complicated with multiple paths, but I also know that @kalman is going to be doing even more work in here to enable more cores/parallelization #1972 in the next few days/weeks and I will be coming back to remove the commit options #2100 and nested maps #1956.  This code will be refactored more soon.  It is good you saw the complexity and pointed out another option, I am just picking one of two evils and will go with the current version for now. It is bugging me so I might come back and refactor this file as a separate PR now.